### PR TITLE
Small API fixes

### PR DESCRIPTION
**Pull request by cmelchior**
Fixes to RealmList.remove(), RealmList.first() and RealmList.last() found during review + additional unit tests to capture errors.

@bmunkholm @kneth @emanuelez

### CODE REVIEW

File: realm/src/androidTest/java/io/realm/RealmListTest.java [hunk 1]
```diff
@@ -211,6 +211,35 @@ public void testMoveUp_nonManagedMode() {
...
+        RealmList<AllTypes> list = new RealmList<AllTypes>();
+        AllTypes object1 = new AllTypes();
+        AllTypes object2 = new AllTypes();
+        list.add(object1);
+        list.add(object2);
+
+        assertEquals(object1, list.first());
+    }
+
+    public void testFirstEmptyList_nonManagedMode() {
```
Comment (bmunkholm) [inline realm/src/androidTest/java/io/realm/RealmListTest.java]: I would merge this with the other testing last. So having the empty list, verify that all methods behaves correctly (also size()).

File: realm/src/androidTest/java/io/realm/RealmListTest.java [hunk 2]
```diff
@@ -294,29 +323,48 @@ public void testGetObjects() {
...
+    }
+
+    public void testLast() {
         Owner owner = testRealm.where(Owner.class).findFirst();
         Dog dog = owner.getDogs().last();
 
         assertEquals("Dog " + (TEST_OBJECTS - 1), dog.getName());
     }
 
+    public void testLastEmptyList() {
```
Comment (bmunkholm) [inline realm/src/androidTest/java/io/realm/RealmListTest.java]: merge with testFirstEmptyList. 
Too much duplicate code. Same as above comment, but here the benefits are even more clear :-)

**Comment by emanuelez** [comment]
:+1: I actually like the atomicity of the unit tests

**Comment by cmelchior** [comment]
That would be my concern as well, we sacrifice transparency/readability for a smaller codebase/faster test execution.

**Comment by kneth** [comment]
:+1:

**Comment by cmelchior** [comment]
@bmunkholm I noticed a few native crashes when doing the empty list unit tests. And I just checked the LinkView JNI code. At least get() and set() doesn't validate their input properly, hence the Java checks. So yes, we should probably do another roll on the RealmList.

**Pull request merged**